tests/config/test_config.py

```diff
@@ -43,10 +58,14 @@ def test_config_get_processes_depended_on_values(
 
 
 def generate_environment_variable_tests() -> Iterator[Tuple[str, str, str, bool]]:
-    for env_value, value in [("true", True), ("false", False)]:
-        for name in get_boolean_options():
-            env_var = "POETRY_{}".format(re.sub("[.-]+", "_", name).upper())
-            yield name, env_var, env_value, value
+    for normalizer, values in [
+        (boolean_normalizer, [("true", True), ("false", False)]),
+        (int_normalizer, [("4", 4), ("2", 2)]),
+    ]:
+        for env_value, value in values:
+            for name in get_options_based_on_normalizer(normalizer=normalizer):
+                env_var = "POETRY_{}".format(re.sub("[.-]+", "_", name).upper())
```

Comment:
Let's use `string + concat` here instead of `.format()`.

Reply:
`f-Strings` should also work fine right? Or do you prefer `+`?

Reply:
I generally try to avoid a f-string where we're adding a single character and have some inline string transformation -- `+ concat` is just more readable with less syntax.